### PR TITLE
feat: [SRM-11287]: Testcases Datadog template

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by deepesh-ui** [comment]
retrigger titlecheck

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10310)

File: src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/PrometheusHealthSource.utils.ts [hunk 1]
```diff
@@ -355,7 +355,7 @@ export function transformPrometheusHealthSourceToSetupSource(
           getString('cv.monitoringSources.prometheus.prometheusMetric'),
           {
             metricName: getString('cv.monitoringSources.prometheus.prometheusMetric'),
-            isManualQuery: false,
+            isManualQuery: isTemplate ? true : false,
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/PrometheusHealthSource.utils.ts:358]: shall we give it as `Boolean(isTemplate)`?

File: src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/tests/PrometheusHealthSourceTemplate.test.tsx [hunk 2]
```diff
@@ -10,25 +10,16 @@ import { Container } from '@wings-software/uicore'
...
-          className="manualQuery"
-          onClick={() => {
-            props.onChange(PrometheusMonitoringSourceFieldNames.IS_MANUAL_QUERY, true)
-          }}
-        />
-      </Container>
-    )
-  }
-}))
+// import { PrometheusMonitoringSourceFieldNames } from '../PrometheusHealthSource.constants'
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/tests/PrometheusHealthSourceTemplate.test.tsx:16]: remove this comment

File: src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/PrometheusHealthSource.utils.ts (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/PrometheusHealthSource.utils.ts]: done

File: src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/tests/PrometheusHealthSourceTemplate.test.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/PrometheusHealthSource/tests/PrometheusHealthSourceTemplate.test.tsx]: done

**Review by praneshg239 — APPROVED** [review]

**Review by amitsh14 — APPROVED** [review]

**Pull request merged**